Src/WindowsServer/WindowsServer.Shared/Implementation/DataContracts/AzureInstanceComputeMetadata.cs

```diff
@@ -0,0 +1,196 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation.DataContracts
+{
+    using System;
+    using System.Globalization;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// Class representing the returned structure from an Azure Instance Metadata request
+    /// for Compute information.
+    /// </summary>
+    [DataContract]
+    internal class AzureInstanceComputeMetadata
+    {
+        private const int ResourceGroupNameLengthMax = 90;
+        private const int ResourceGroupNameLengthMin = 1;
+        private const string ResourceGroupNameValidChars = @"^[a-zA-Z0-9\.\-_]+$";
```

Comment:
Do you really need to use RegExp for this? If this is called a lot (I think it is), then consider a zero-allocation static helper method.
e.g.
```C#
private static bool IsValidResourceGroupName(string s)
{
    if (string.IsNullOrEmpty(s)) return false;
    foreach(var ch in s)
    {
        if (!IsValidResourceGroupNameChar(ch)) return false;
    }

    return true;
}

private static bool IsValidResourceGroupNameChar(char ch)
{
    // todo
}
```

Reply:
This is not called a lot. It is called exactly once each when Azure IMS service is present upon app startup (in a 'fire and forget' manner). Which means each of these regex calls happen exactly once each for a successful Azure IMS request.

Reply:
OK, then.